flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecSink.java

```diff
@@ -43,20 +43,25 @@ public BatchExecSink(
             String description) {
         super(
                 tableSinkSpec,
-                tableSinkSpec.getTableSink().getChangelogMode(ChangelogMode.insertOnly()),
                 true, // isBounded
                 getNewNodeId(),
                 Collections.singletonList(inputProperty),
                 outputType,
                 description);
     }
 
+    @Override
+    protected ChangelogMode getOutputChangelogMode(PlannerBase planner) {
```

Comment:
a sink does not have an output, but a `getConsumedChangelogMode`, do we really need this method? see comment below.